File: onyx-database-tests/src/entities/relationship/OneToOneThreeDeep.java
package entities.relationship;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.annotations.*;
import entities.AbstractEntity;

/**
 * Created by timothy.osborn on 1/26/15.
 */
@Entity
public class OneToOneThreeDeep extends AbstractEntity implements IManagedEntity
{
    @Attribute
    @Identifier
    public long id;

    @Relationship(type = RelationshipType.ONE_TO_ONE, inverseClass = OneToOneRecursiveChild.class, inverse = "third")
    public OneToOneRecursiveChild parent;

}

File: onyx-database-tests/src/main/kotlin/entities/relationship/OneToOneThreeDeep.kt
package entities.relationship

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.annotations.*
import com.onyx.persistence.annotations.values.RelationshipType
import entities.AbstractEntity

/**
 * Created by timothy.osborn on 1/26/15.
 */
@Entity
class OneToOneThreeDeep : AbstractEntity(), IManagedEntity {
    @Attribute
    @Identifier
    var id: Long = 0

    @Relationship(type = RelationshipType.ONE_TO_ONE, inverseClass = OneToOneRecursiveChild::class, inverse = "third")
    var parent: OneToOneRecursiveChild? = null

}
